simple_history/management/commands/_populate_utils.py

```diff
@@ -15,15 +16,45 @@ def get_history_model_for_model(model):
     return getattr(model, manager_name).model
 
 
-def bulk_history_create(model, history_model, batch_size):
-    """Save a copy of all instances to the historical model."""
-    historical_instances = [
-        history_model(
-            history_date=getattr(instance, '_history_date', now()),
-            history_user=getattr(instance, '_history_user', None),
-            **{
-                field.attname: getattr(instance, field.attname)
-                for field in instance._meta.fields
-            }
-        ) for instance in model.objects.all()]
-    history_model.objects.bulk_create(historical_instances, batch_size=batch_size)
+def bulk_history_create(stdout, model, history_model, batch_size, max_in_memory_models=5000):
+    """Save a copy of all instances to the historical model.
+        :param max_in_memory_models: number of models to load into memory at once. This prevents
+            memory issues from having too many model instances in memory
+    """
+
+    historical_instances = []
+    count = 1
+    chunk_count = 0
+    for instance in model.objects.iterator():
+        if count > max_in_memory_models:
+            count = 0
+            start_num = chunk_count * max_in_memory_models
+            end_num = start_num + max_in_memory_models
+            stdout.write(
+                "Starting bulk creating history models for {} instances {}-{}"
```

Comment:
maybe delete or hide behind a flag these prints its a lot of logging if its low and you have a lot of models. The batch doesn't print.